### PR TITLE
Expand godot docs

### PR DESCRIPTION
**Pull request by dogboydog**
Add more Godot docs describing most components of the plugin. Most of it is a modified version of the Unity docs (since the Godot plugin is based on the Unity plugin code), but I tried to avoid any inaccuracies.

### CODE REVIEW

**Review by ptlthg — COMMENTED** [review]
Here are some small corrections I had while reading through this, looks good overall though. Thanks for improving the docs

File: docs/using-yarnspinner-with-godot/components/dialogue-views/custom-dialogue-views.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+---
+
+While the Line View and Options List View are useful for lots of situations, your game might need to display lines and options in specific ways. In these situations, you can write your own custom Dialogue View, and handle the presentation of lines and options in ways that are entirely in your control.
+
+## Creating a Dialogue View
+
+To create a Dialogue View, you write a script that implements the `DialogueViewBase` interface, and add your script to a node in your scene. You can then add this node to the Dialogue Views list on your scene's Dialogue Runner.
+
+{% hint style="info" %}
+If you just want to skip straight to the sample code, take a look at the [`SimpleSpeechBubbleLineView` sample code](https://github.com/YarnSpinnerTool/ExampleProjects/blob/main/UtilityScripts/SimpleSpeechBubbleLineView.cs) in the Yarn Spinner examples repository.
```
[the remark this answers is not in the document]
Comment (dogboydog) [reply docs/using-yarnspinner-with-godot/components/dialogue-views/custom-dialogue-views.md:12]: Oops. I meant to just remove this part, I'll do so

File: docs/using-yarnspinner-with-godot/components/dialogue-views/line-view.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+| Fade Out Time| The duration of the Fade effect when fading a line out, in seconds. If this is zero, the line will disappear immediately.|
+| Use Typewriter Effect| If this is turned on, the text of the line will appear one character at a time. This will take place after the Fade effect, if enabled.|
+| On Character Typed| A signal that's emitted every time the Typewriter effect displays new text.|
+| Typewriter Effect Speed| The number of characters per second to display when performing a Typewrite effect. Larger values means that text will appear faster.|
+| Character Name Text| A RichTextLabel node that will display the name of the character currently speaking the line.|
+| Show Character Name In Line View| If this is turned on, lines that contain a character's name will display the name in the Line Text section. If it is turned off, character names will not be shown at all. This option is only available when Character Name Text is empty.|
+| Continue Button| A Control that will be made visible when the line has finished appearing. This is intended to be used for controlling the appearance of a button that the user can interact with to continue to the next line.|
```
[the remark this answers is not in the document]
Comment (dogboydog) [reply docs/using-yarnspinner-with-godot/components/dialogue-views/line-view.md:53]: Good point. I should add an article about MarkupPallete

File: docs/using-yarnspinner-with-godot/creating-commands-functions.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+public partial class CustomWaitCommand : Node{    
+
+    [YarnCommand("custom_wait")]
+    static async Task CustomWait() {
+
+        // Wait for 1 second
+        var mainLoop = Engine.GetMainLoop();
+        var sceneTree = mainLoop as SceneTree;
+        var timer = sceneTree.CreateTimer(1.0);
+        await mainLoop.ToSignal(timer, "timeout");
```
[the remark this answers is not in the document]
Comment (dogboydog) [reply docs/using-yarnspinner-with-godot/creating-commands-functions.md:151]: Nice, good catch

**Pull request merged**